### ISSUE TITLE
CustomJoint (Glenohumeral) might be treated incorrectly

### ISSUE THREAD

**Issue by itbellix** [itbellix/opensimAD#4]
When using my custom version of OpenSimAD I run into issues, as the x_dot computed through the CasADi external function and the one computed with the more classic OpenSim  `realizeAcceleration()` are different.
I suspect this could be caused by the way in which the `Glenohumeral` joint is currently implemented (when running Antoine's code for creating the model from scratch in the .cpp file). I think it would be more efficient in my case to simply load the already existing model into cpp.

**Comment by itbellix** [comment]
Currently, the GH joint is transformed into cpp code as follows:

```cpp
// Definition of joints.
	OpenSim::WeldJoint* scapula_to_ground;
	scapula_to_ground = new OpenSim::WeldJoint("scapula_to_ground", model->getGround(), Vec3(0.00000000000000000000, 0.00000000000000000000, 0.00000000000000000000), Vec3(-0.00000000000000000000, 0.00000000000000000000, -0.00000000000000000000), *scapula, Vec3(0.00000000000000000000, 0.00000000000000000000, 0.00000000000000000000), Vec3(0.00000000000000000000, 0.00000000000000000000, 0.00000000000000000000));
	model->addJoint(scapula_to_ground);

	SpatialTransform st_GlenoHumeral;
	st_GlenoHumeral[0].setCoordinateNames(OpenSim::Array<std::string>("plane_elv", 1, 1));
	st_GlenoHumeral[0].setFunction(new LinearFunction(1.0000, 0.0000));
	st_GlenoHumeral[0].setAxis(Vec3(0.00000000000000000000, 1.00000000000000000000, 0.00000000000000000000));
	st_GlenoHumeral[1].setCoordinateNames(OpenSim::Array<std::string>("shoulder_elv", 1, 1));
	st_GlenoHumeral[1].setFunction(new LinearFunction(1.0000, 0.0000));
	st_GlenoHumeral[1].setAxis(Vec3(-1.00000000000000000000, 0.00000000000000000000, 0.00000000000000000000));
	st_GlenoHumeral[2].setCoordinateNames(OpenSim::Array<std::string>("axial_rot", 1, 1));
	st_GlenoHumeral[2].setFunction(new LinearFunction(1.0000, 0.0000));
	st_GlenoHumeral[2].setAxis(Vec3(-0.08459999999999999465, 0.99470000000000002860, -0.05840000000000000052));
	st_GlenoHumeral[3].setFunction(new MultiplierFunction(new Constant(0.00000000000000000000), 0.87167774563804756305));
	st_GlenoHumeral[3].setAxis(Vec3(1.00000000000000000000, 0.00000000000000000000, 0.00000000000000000000));
	st_GlenoHumeral[4].setFunction(new MultiplierFunction(new Constant(0.00000000000000000000), 0.96383680480477873509));
	st_GlenoHumeral[4].setAxis(Vec3(0.00000000000000000000, 1.00000000000000000000, 0.00000000000000000000));
	st_GlenoHumeral[5].setFunction(new MultiplierFunction(new Constant(0.00000000000000000000), 0.95484232176803507386));
	st_GlenoHumeral[5].setAxis(Vec3(0.00000000000000000000, 0.00000000000000000000, 1.00000000000000000000));
	OpenSim::CustomJoint* GlenoHumeral;
	GlenoHumeral = new OpenSim::CustomJoint("GlenoHumeral", *scapula, Vec3(-0.00832452451869142816, -0.03277044672940921527, 0.00859358429549605951), Vec3(0.00000000000000000000, 0.00000000000000000000, 0.00000000000000000000), *humerus, Vec3(0.00000000000000000000, 0.00000000000000000000, 0.00000000000000000000), Vec3(0.00000000000000000000, 0.00000000000000000000, 0.00000000000000000000), st_GlenoHumeral);
	model->addJoint(GlenoHumeral);
```

This is very inefficient imho, and prone to errors (compare [this issue](https://github.com/opensim-org/opensim-core/issues/2066) on opensim-core).

It would be desireable to substitute all of this with a couple of lines that just load the model from the file we have, since in my case there is no need to define a new model programmatically.

**Comment by itbellix** [comment]
The real issue I was having when calculating accelerations was actually due to the way in which the function `setStateVariableValues(*state, QsUs)` works. The correct order to call my CasADi function is providing it with the state in the form of `[q_1, q_1_dot, q_2, q_2_dot, ...]`.
This is solved in a733c6db5b4b626f59e4c9fb21c834c05b844b98.

Another thing to keep in mind is the [F_map](https://github.com/itbellix/opensimAD/blob/main/utilities.py#L1153-L1168) object that is created exactly to solve this problem.
